### PR TITLE
Add production-like Run 3 HI MC workflow to relval

### PR DESCRIPTION
**Pull request by mandrenguyen**
#### PR description:

Upon the suggestion of @dpiparo we would like include heavy-ion reconstruction in our profiling.  To do this with our profiling machinery, we need a "production-like" heavy-ion MC workflow, i.e., not including validation.
This PR adds such a workflow (159.03), and cleans up the numbering scheme of HI workflows slightly.

#### PR validation:

Tested with new workflow 159.03, and ensured that 159.0 and 159.02 were not broken by the migration. 

A backport to 12_5_X will probably be useful.

### CODE REVIEW

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 1]
```diff
@@ -2806,6 +2807,19 @@ def gen2021HiMix(fragment,howMuch):
                                                           '--procModifiers':'genJetSubEvent',
                                                         },step3Up2015Defaults])
 
+steps['RECOHI2022PROD']=merge([hiDefaults2021_ppReco_approxClusters,{'-s':'RAW2DIGI,L1Reco,RECO',
+                                                                     '--datatier':'AOSSIM',
```
Comment (missirol) [inline Configuration/PyReleaseValidation/python/relval_steps.py:2811]: AODSIM?

**Comment by perrotta** [comment]
please test

**Comment by srimanob** [comment]
test parameters:
- workflows = 158.5,159.0,159.02,159.03

**Comment by srimanob** [comment]
test parameters:
- workflows = 158.5,159.0,159.02,159.03
- relvals_opt = --what cleanedupgrade,standard,highstats,pileup,generator,extendedgen,production,ged,machine,premix

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by srimanob** [comment]
+Upgrade

Additional workflows run fine.

**Comment by mandrenguyen** [comment]
@cms-sw/pdmv-l2 just a gentle ping

**Comment by kskovpen** [comment]
+pdmv

**Comment by perrotta** [comment]
+1

**Pull request merged**